src/wrappers/query.ts

```diff
@@ -0,0 +1,37 @@
+import {ServiceOptions} from '../types'
+import {isInBrowser} from '../utils/platform'
+import {Stream} from 'stream'
+
+export default class {
+  private basePath: string
+  private baseOptions: ServiceOptions
+
+  constructor(basePath: string, baseOptions: ServiceOptions) {
+    this.basePath = basePath
+    this.baseOptions = baseOptions
+  }
+
+  public execute(
+    orgID: string,
+    query: string,
+    extern?: File
+  ): {stream: Stream; cancel: () => void} {
```

Comment:
If this returns a stream `out` that has a method `out.destroy`, does it also need to return the `cancel` function? Seems redundant.

Reply:
I think its better to keep the concerns separate for canceling a request and destroying the stream, to merge them we would have to override destroy which seems like a bad idea.

Reply:
Ok, that seems reasonable to me.

I think what makes me hesitate is that streams feel like an abstraction for programming in the functional reactive style, but calling a cancellation method is at odds with that paradigm. I think in the FRP style, cancellation happens implicitly when there are no more subscribers to the stream. But also FRP is a total rabbit hole that doesn't integrate well non-FRP codebases so w/e 😸